server/GlideRecord.d.ts

```diff
@@ -1,5 +1,41 @@
 import { SNAPIGlideRecord } from './SNAPIGlideRecord';
-declare class GlideRecord extends SNAPIGlideRecord {
+import { GlideQueryCondition } from './GlideQueryCondition';
+import { GlideElement } from './GlideElement';
+import { QueryOperator } from '../util';
+type FieldType<T> = Extract<keyof T, string>;
```

Comment:
Are you using `Extract` on purpose? It seems almost identical to `keyof T`, with the only difference when the type has numbers as fields:
```typescript
interface my_table {
  test: string;
  test2: number;
  1: boolean;
}

type FieldType1<T> = Extract<keyof T, string>;
type FieldType2<T> = keyof T;

type MyFieldType1 = FieldType1<my_table>; // = "test" | "test2"
type MyFieldType2 = FieldType2<my_table>;  // = "test" | "test2" | 1
```

Not sure if there is something ServiceNow specific here, or is it actually possible to have numbers as column names. Just curious :)

Reply:
I answered this myself.
If we use `keyof T` we cannot extend the base SNAPI class cause original arguments are only strings, not numbers:
`Type 'string' is not assignable to type 'keyof T'`